config/environments/test.rb

```
@@ -92,7 +92,7 @@
   config.force_ssl = false
 
   # Enable asset compilation for system tests
-  if ENV['RUN_RAILS_SYSTEM_TESTS'] == 'true'
+  if ENV['RUN_RAILS_SYSTEM_TESTS'] == 'true' || ENV['RUN_SYSTEM_TESTS'] == 'true'
```

Adding this, per claude's recommendation, resolved some cold-start issues I was having locally where the first test would always fail.

I'm still not totally sure of the root problem, or why this fixes it. But what I think might be going on is: with the recent addition of rails system tests (in addition to existing react system tests), the rails server is newly serving frontend assets, and here we tell the server _not_ to precompile those assets, so it doesn't take forever to start up.

I'd be interested to know whether other developers also encounter this when running hmis system tests locally (i.e. on release branch, not using the change I added here). And interested if @eanders you could perhaps provide more insight about this config - is my guess on the right track or totally off base?

---

My goal with the separate ENV (if memory serves) was simply to allow us to run one set of system tests in isolation.  It's totally possible I introduced some unexpected interaction with the existing system tests by doing so.  
I wasn't running the front-end (react) tests locally, except once or twice to confirm they worked early on.  I have not seen them fail early on GitHub Actions.

---

Thanks @eanders that makes sense! Sorry I should have clarified, by "this config" what I really meant was lines 96-98 in this file. 
```ruby
    config.assets.compile = true
    config.assets.check_precompiled_asset = false
    config.assets.digest = false
```

I don't really know what they are doing, but adding them to the react system tests (as well as the rails ones) seems to have fixed the local cold start issues I was seeing.

---

My understanding of 96-98 is that tells rails that we should compile the assets, but don't check to confirm they are precompiled, and don't rely on (or maybe create) a digest. 
My read of that is basically, just trust that the assets are correct.